CompoundV3-Bots/Transfers-Monitor/src/agent.ts

```diff
@@ -0,0 +1,112 @@
+import {
+  Finding,
+  Initialize,
+  HandleTransaction,
+  TransactionEvent,
+  ethers,
+} from "forta-agent";
+import { NetworkManager } from "forta-agent-tools";
+import { NetworkData, TransferLog } from "./utils";
+import { BUY_COLLATERAL_ABI, SUPPLY_ABI, TRANSFER_ABI } from "./constants";
+import CONFIG from "./agent.config";
+import { BigNumber, getDefaultProvider } from "ethers";
+import { createTransferFinding } from "./finding";
+
+const networkManager = new NetworkManager(CONFIG);
+
+export const provideInitialize = (
+  networkManager: NetworkManager<NetworkData>,
+  provider: ethers.providers.Provider
+): Initialize => {
+  return async () => {
+    await networkManager.init(provider);
+  };
+};
+
+export const provideHandleTransaction = (
+  networkManager: NetworkManager<NetworkData>
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const transferEvents: Set<any> = new Set([]);
```

Comment:
Why a `Set`? In this case it's not necessary IMO, and if so, filling it with objects will basically make it a set for the references/pointers.

Reply:
Oh, I got why later, but take a look at my later comments